File: MySQL-MariaDB/Ruby/Select/mysql_select.rb
require 'mysql2'

begin
 puts "Connecting to MySQL database..."
 client = Mysql2::Client.new(:host => "localhost", :username => "root", :password => "My$ql@Server#5.7", :database => "classicmodels") 
 if client.ping
  puts "...connection established."
 else
  puts "...connection failed."
 end
 # results = client.query("SELECT * FROM customers ORDER BY customerNumber DESC LIMIT 2;")
 results = client.query("SELECT * FROM customers;")
 if !results.nil?
  # puts results.count
  results.each do |row|
   counter = 0
   row.each do |column|
    x = column[1]
    if x.nil?
     print "nil" 
    elsif x.class.equal? BigDecimal
      print "%.2f" % x
    else
      print x
    end
    counter = counter + 1
    if counter <  row.count
     print " - "
    end
   end
   puts 
  end
 end
rescue Mysql2::Error => e
 puts e.errno
 puts e.error
ensure
 if !client.nil?
  client.close 
  puts "Connection closed."
 end
end

File: MySQL-MariaDB/Perl/Select/mysql_select.pl
use feature ':5.22';
use strict;
use warnings;
# Git CMD # > ppm install DBD-mysql
use DBI;

=pod
sub stringify_undef_for {
    for my $x ( @_ ) {
        if (! defined $$x) {
            $$x = "undef";            
        }
    }
}
=cut

# MySQL
my $dsn = "DBI:mysql:classicmodels";
my $username = "root";
my $password = 'My$ql@Server#5.7';

my $dbh = DBI->connect($dsn, $username, $password);

die "failed to connect to MySQL database:DBI->errstr()" unless($dbh);

say "Connected to MySQL Server successfully."; 

# my $sth = $dbh->prepare("SELECT * FROM customers ORDER BY customerNumber DESC LIMIT 2;") or die "prepare statement failed: $dbh->errstr()";

my $sth = $dbh->prepare("SELECT * FROM customers;") or die "prepare statement failed: $dbh->errstr()";
 
$sth->execute() or die "execution failed: $dbh->errstr()"; 
 
$sth->dump_results();

=pod
# my ($customerNumber, $customerName, $contactLastName, $contactFirstName, $phone, $addressLine1, $addressLine2, $city, $state, $postalCode, $country, $salesRepEmployeeNumber, $creditLimit);
 
while(my ($customerNumber, $customerName, $contactLastName, $contactFirstName, $phone, $addressLine1, $addressLine2, $city, $state, $postalCode, $country, $salesRepEmployeeNumber, $creditLimit) = $sth->fetchrow()) {
    # no warnings 'uninitialized';
    stringify_undef_for(\$customerNumber, \$customerName, \$contactLastName, \$contactFirstName, \$phone, \$addressLine1, \$addressLine2, \$city, \$state, \$postalCode, \$country, \$salesRepEmployeeNumber, \$creditLimit);
    say("$customerNumber - $customerName - $contactLastName - $contactFirstName - $phone - $addressLine1 - $addressLine2 - $city - $state - $postalCode - $country - $salesRepEmployeeNumber - $creditLimit");                   
    # say($customerNumber . " - " . $customerName . " - " . $contactLastName . " - " . $contactFirstName . " - " . $phone . " - " . $addressLine1 . " - " . $addressLine2 . " - " . $city . " - " . $state . " - " . $postalCode . " - " . $country . " - " . $salesRepEmployeeNumber . " - " . $creditLimit);                   
}
=cut

$sth->finish();

$dbh->disconnect();